daemon/logger/adapter.go

```diff
@@ -0,0 +1,115 @@
+package logger
+
+import (
+	"encoding/json"
+	"io"
+	"os"
+	"sync"
+
+	"github.com/docker/docker/pkg/plugingetter"
+	"github.com/pkg/errors"
+)
+
+// pluginAdapter takes a plugin and implements the Logger interface for logger
+// instances
+type pluginAdapter struct {
+	driverName   string
+	id           string
+	plugin       logPlugin
+	fifoPath     string
+	stream       io.WriteCloser
+	capabilities Capability
+	mu           sync.Mutex // synchronize access to the log stream
+	logInfo      Info
+
+	// TODO(cpuguy83): not JSON, how about protobuf?
+	// Just using json for quick and dirty implementation
+	// here we are just caching the encoder for writing to the fifo
+	enc *json.Encoder
```

Comment:
I think it might be a time to look for another serialization format. Protobuf is quite ubiquitous now, looks like a good choice.